src/helpers/credential.rs

```diff
@@ -269,6 +279,138 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    pub async fn create_self_issued_credential(
+        &mut self,
+        schema_did: &str,
+        credential_subject_str: &str,
+        bbs_secret: &str,
+        bbs_private_key: &str,
+        credential_revocation_did: Option<&str>,
+        credential_revocation_id: Option<&str>,
+        exp_date: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential_subject: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let credential_subject_copy: CredentialSubject = serde_json::from_str(credential_subject_str)?;
```

Comment:
Should we use 2 variables for the same thing? :D Could you just use `.clone()` where needed directly? `credential_subject.clone()`

Reply:
It is a leftover from many experiments which I did, trying to make finish_credential() work :). Will cleanup the code as soon as it works.

Reply:
Done